devito/operator.py

```diff
@@ -154,6 +155,9 @@ def __init__(self, expressions, **kwargs):
         # autotuning reports, etc
         self._state = {}
 
+        # Form and gather any required implicit expressions
```

Comment:
I think it's worth spending a few words about what an implicit expression is and, in particular, why does this take place *before* the _expression lowering_ pass